modules/users/tests/server/user.server.routes.tests.js

```diff
@@ -262,6 +262,601 @@ describe('User CRUD tests', function () {
     });
   });
 
+  it('forgot password should return 400 for non-existent username', function (done) {
+    user.roles = ['user'];
+
+    user.save(function (err) {
+      should.not.exist(err);
+      agent.post('/api/auth/forgot')
+        .send({
+          username: 'some_username_that_doesnt_exist'
+        })
+        .expect(400)
+        .end(function (err, res) {
+          // Handle error
+          if (err) {
+            return done(err);
+          }
+
+          res.body.message.should.equal('No account with that username has been found');
+          return done();
+        });
+    });
+  });
+
+  it('forgot password should return 400 for no username provided', function (done) {
+    var provider = 'facebook';
+    user.provider = provider;
+    user.roles = ['user'];
+
+    user.save(function (err) {
+      should.not.exist(err);
+      agent.post('/api/auth/forgot')
+        .send({
+          username: ''
+        })
+        .expect(400)
+        .end(function (err, res) {
+          // Handle error
+          if (err) {
+            return done(err);
+          }
+
+          res.body.message.should.equal('Username field must not be blank');
+        });
+
+        agent.post('/api/auth/forgot')
+          .expect(400)
+          .end(function (err, res) {
+            // Handle error
+            if (err) {
+              return done(err);
+            }
+
+            res.body.message.should.equal('Username field must not be blank');
+            return done();
+          });
+
+    });
+  });
+
+  it('forgot password should return 400 for non-local provider set for the user object', function (done) {
+    var provider = 'facebook';
+    user.provider = provider;
+    user.roles = ['user'];
+
+    user.save(function (err) {
+      should.not.exist(err);
+      agent.post('/api/auth/forgot')
+        .send({
+          username: user.username
+        })
+        .expect(400)
+        .end(function (err, res) {
+          // Handle error
+          if (err) {
+            return done(err);
+          }
+
+          res.body.message.should.equal('It seems like you signed up using your ' + user.provider + ' account');
+          return done();
+        });
+    });
+  });
+
+  it('forgot password should return 400 for user password reset request', function (done) {
+    // General comment about this test-
```

Comment:
I think for the purposes of these tests, we should only be concerned with the responses from the CRUD operations. Not the actual sending of the emails. 

I think it would be too complicated to test that an email was actually sent. If we did this, then we'd end up including the testing of a service not related to the CRUD operations here. We don't want to mix test cases like this.

If we wanted to test the email functionality, they should be isolated into their own tests.